### PR TITLE
Android 13

### LINKED ISSUE #7236: [Android 13] Set 'compileSdk' and `targetSdk` to 33
More details: https://developer.android.com/about/versions/13

Some dependency upgrade are requiring this (for instance #7074)

Please also consider investigating this issue: https://github.com/vector-im/element-android/issues/6379

Previous similar task was in #6863.

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Migration to the new back button management is not done, will be handled separately (change will be tracked by #7313). Still using (but ignoring warning) deprecated methods about it.

New permission to display notification ("android.permission.POST_NOTIFICATIONS") has been added.

Closes #7236 

## Motivation and context

Compile with Android 13 sdk and target Android 13.

## Tests

- Test with android 13 emu

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt [hunk 1]
```diff
@@ -113,11 +113,11 @@ internal fun ViewPropertyAnimator.setAnimatorListener(
         onAnimationStart: ((Animator?) -> Unit)? = null
```
Comment (Florian14) [inline library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt:113]: You can now remove the nullability on the `Animator` and probably update the usage

Comment (bmarty) [reply library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt:113]: ec037270b54baef56f5631bd81221432cb69ea24

Comment (bmarty) [reply library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt]: c17b55b874daa656a2ee295b372375adf841ad73

File: library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt [hunk 2]
```diff
@@ -96,31 +96,27 @@ class SwipeToDismissHandler(
...
                 .start()
     }
 }
 
-internal fun ViewPropertyAnimator.setAnimatorListener(
-        onAnimationEnd: ((Animator?) -> Unit)? = null,
-        onAnimationStart: ((Animator?) -> Unit)? = null
-) = this.setListener(
+private fun ViewPropertyAnimator.setAnimatorEndListener(
+        onAnimationEnd: () -> Unit,
```
Comment (Florian14) [inline library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt:112]: Not a blocker for the PR but don't you think you should pass the `Animator` inside the lambda, even if it is not used? (Just to keep the API behaviour)

File: library/multipicker/src/main/java/im/vector/lib/multipicker/Picker.kt [hunk 3]
```diff
@@ -91,6 +94,7 @@ abstract class Picker<T> {
         } else if (dataUri != null) {
             selectedUriList.add(dataUri)
         } else {
+            @Suppress("DEPRECATION")
```
Comment (Florian14) [inline library/multipicker/src/main/java/im/vector/lib/multipicker/Picker.kt:97]: why did you add the deprecation here?

File: vector/src/main/java/im/vector/app/core/extensions/MavericksViewModel.kt [hunk 4]
```diff
@@ -25,6 +25,7 @@ import com.airbnb.mvrx.MavericksViewModelProvider
 
 inline fun <reified VM : MavericksViewModel<S>, reified S : MavericksState> ComponentActivity.lazyViewModel(): Lazy<VM> {
     return lazy(mode = LazyThreadSafetyMode.NONE) {
+        @Suppress("DEPRECATION")
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/core/extensions/MavericksViewModel.kt:28]: why did you add the deprecation here?

**Review by Florian14 — APPROVED** [review]
The changes look fine, just two small questions.
Not tested

File: library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt (hunk 2 above)
Comment (bmarty) [reply library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt:112]: it's just a private extension, so let's simplify it.

File: library/multipicker/src/main/java/im/vector/lib/multipicker/Picker.kt (hunk 3 above)
Comment (bmarty) [reply library/multipicker/src/main/java/im/vector/lib/multipicker/Picker.kt:97]: `get` is deprecated, but since we check for multiple types, I cannot use the new API.

File: vector/src/main/java/im/vector/app/core/extensions/MavericksViewModel.kt (hunk 4 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/core/extensions/MavericksViewModel.kt:28]: same answer.

**Comment by bmarty** [comment]
Thanks for the review. Let's merge the PR to make it available in the next nightly.
